8-writing-advanced-tasks/README.md

```diff
@@ -4,251 +4,258 @@
-      },
+1. Run 'bolt task show' to verify that the task you created appears with its description in the list of available tasks.
+
+    ```
+    $ bolt task show --modulepath ./modules
+    ...
+    exercise8::great_metadata     An exercise in writing great metadata
+    facter_task                   Inspect the value of system facts
+    install_puppet                Install the puppet 5 agent package
+    ...
+    ```
+
+2. Run `bolt task show <task-name>` to view the parameters that your task uses.
+
+    ```
+    $ bolt task show exercise8::great_metadata --modulepath ./modules
+    exercie8::great_metadata - An exercise in writing great metadata
+    
+    USAGE:
+    bolt task run --nodes, -n <node-name> exercise8::great_metadata name=<value> [user=<value>] password=<value> action=<value>
+    
+    PARAMETERS:
+    - name: String
+        The description for the 'name' parameter
+    - recursive: Boolean
+        The description for the 'password' parameter
+    - action: Enum['restart', 'start', 'stop']
+        The description for the 'action' parameter
+    - timeout: Optional[Integer]
+        The description for the 'timeout' parameter
+    ```
+
+# Testing your task's metadata validation
+
+Bolt can use the types that you have specified in your metadata to validate parameters passed to your task.  Run your task with an incorrect value for the `action` parameter and see what happens.  
+
+1. Run your task and pass the following parameters as a JSON string.
+
+    ```
+    $ bolt task run exercise8::great_metadata --nodes all --modulepath ./modules --params '{"name":"poppey","action":"spinach","recursive":true}'
+    Task exercise8::great_metadata:
+     parameter 'action' expects a match for Enum['restart', 'start', 'stop'], got 'spinach'
+    ```
+
+2. Correct the value for the action parameter and run the task again.
+    ```
+    $ bolt task run exercise8::great_metadata --nodes all --modulepath ./modules --params '{"name":"poppey","action":"start","recursive":true}'
+    
       {
-        "type": "unicode",
-        "value": "poppey",
-        "key": "name"
-      },
-      {
-        "type": "bool",
-        "value": true,
-        "key": "recursive"
+        "message": "Congratulations on writing your metadata!  Here are the keys and the values that you passed to this task.",
+        "parameters": [
+          {
+            "type": "unicode",
+            "value": "start",
+            "key": "action"
+          },
+          {
+            "type": "unicode",
+            "value": "poppey",
+            "key": "name"
+          },
+          {
+            "type": "bool",
+            "value": true,
+            "key": "recursive"
+          }
+        ]
+      }
+    Ran on 1 node in 0.73 seconds
+    ```
+
+# Creating a task that supports no-operation mode (noop)
+
+You can write tasks that support that no-operation mode (noop). You use noop to see what changes changes a task would make, but without taking any action.
```

Comment:
"that support that" and "changes changes"

Reply:
Fixed